tests/integration/test_id_match.py

```diff
@@ -0,0 +1,214 @@
+                schd, ['1/b2'], ['1'], ['failed'], None
+            )
+        )
+
+        # task pool state:
+        # * cycle 1
+        #   * n=0 a1 waiting
+        #   * n=1 b1 waiting
+        #   * n=2 c1 waiting
+        #   * n=1 a2 succeeded
+        #   * n=0 b2 failed
+        #   * n=1 c2 waiting
+        # * cycle 2
+        #   * n=0 a1 waiting
+        #   * n=1 b1 waiting
+        #   * n=2 c1 waiting
+        #   * n=0 a2 waiting
+        #   * n=1 b2 waiting
+        #   * n=2 c2 waiting
+
+        # check the n=0 window matches expecations before proceeding
+        assert {
+            itask.tokens.relative_id for itask in schd.pool.get_tasks()
+        } == {'1/a1', '1/b2', '3/a1', '3/a2'}
+
+        # test active task selection
+        assert (
+            match('*:waiting')
+            == match('*/root:waiting')
+            == match('*/*:waiting')
+            == match('*/A:waiting')
+            == match('*/a*:waiting')
+            == match('1/a1:waiting', '3/a1:waiting', '3/a2:waiting')
+            == match('^/a1:waiting', '3/a1:waiting', '$/a2:waiting')
+            == ({'1/a1', '3/a1', '3/a2'}, set())
+        )
+
+        assert (
+            match('*:failed')
+            == match('*/root:failed')
+            == match('*/*:failed')
+            == match('*/B:failed')
+            == match('*/b*:failed')
+            == match('1/b2:failed')
+            == match('^/b2:failed')
+            == ({'1/b2'}, set())
+        )
+
+        assert (
+            match('1/b1:failed', '1/b2:failed')
+            == match('1/B:failed', '1/b1:failed')
+            == match('*:failed', '1/B:failed', '1/b1:failed')
+            == ({'1/b2'}, {'1/b1:failed'})
+        )
+
+        assert match('*:submit-failed') == (set(), {'*:submit-failed'})
+
+        # test regular task selection
+        assert (
+            match('*')
+            == match('*/*')
+            == match('*/root')
+            == match('*/A', '*/B', '*/c1', '*/c2')
+            == match('*/a*', '*/b*', '*/c*')
+            == (
+                {
+                    '1/a1',
+                    '1/a2',
+                    '1/b1',
+                    '1/b2',
+                    '1/c1',
+                    '1/c2',
+                    '3/a1',
+                    '3/a2',
+                    '3/b1',
+                    '3/b2',
+                    '3/c1',
+                    '3/c2',
+                },
+                set(),
+            )
+        )
+
+        assert (
+            match('1/A')
+            == match('1/a*')
+            == match('^/a*')
+            == match('1/a1', '1/a2')
+            == ({'1/a1', '1/a2'}, set())
+        )
+
+        assert match('1/X') == (set(), {'1/X'})
+        assert match('5:waiting') == (set(), {'5:waiting'})
+
+        # cycle 2 is not on sequence for task a1
+        assert match('1/a1', '2/a1', '3/a1') == ({'1/a1', '3/a1'}, {'2/a1'})
+
+        # errors
```

Comment:
Can you explain why there is not a check for an error or a warning in the comments?